net/local/local_bind.c

```diff
@@ -52,49 +52,47 @@ int psock_local_bind(FAR struct socket *psock,
   FAR struct local_conn_s *conn;
   FAR const struct sockaddr_un *unaddr =
     (FAR const struct sockaddr_un *)addr;
-  int namelen;
+  size_t namelen;
 
   DEBUGASSERT(psock != NULL && psock->s_conn != NULL &&
-              unaddr != NULL && unaddr->sun_family == AF_LOCAL &&
-              addrlen >= sizeof(sa_family_t));
+              unaddr != NULL && unaddr->sun_family == AF_LOCAL);
+
+  if (addrlen <= sizeof(sa_family_t) + 1)
```

Comment:
Let's change to
```suggestion
  if (addrlen <= sizeof(sa_family_t) + 2)
```
as it seems to be more correct since `unaddr->sun_path[1]` is accessed in `namelen == 0` case and `unaddr->sun_path` will contain at least 2 chars (1 non `'\0'` and one `'\0'`).

Reply:
Seems I was too fast and condition is correct since `<=` handles both cases